src/canmatrix/canmatrix.py

```diff
@@ -779,108 +869,172 @@ def updateReceiver(self):
+            data = newData
+        return self.signals_to_bytes(data)
+
+    def bytes_to_bitstrings(self, data):
+        """Return two arrays big and little containing bits of given data (bytearray)
+
+        :param data: bytearray of bits (little endian). Iterable or bytes.
+            i.e. (0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xA8)
+        :return: bit arrays in big and little byteorder
+        """
+        b = tuple('{:08b}'.format(b) for b in data)
+        little = ''.join(reversed(b))
+        big = ''.join(b)
+
+        return little, big
+
+    def bitstring_to_signal_list(self, signals, big, little):
+        """Return OrderedDictionary with Signal Name: object decodedSignal (flat / without support for multiplexed frames)
+
+        :param signals: Iterable of signals (class signal) to decode from frame.
+        :param big: bytearray of bits (big endian).
+        :param little: bytearray of bits (little endian).
+        :return: array with raw values (same order like signals)
+        """
+        unpacked = []
         for signal in signals:
-            signal_value.append(
-                signal.phys2raw(data.get(signal.name))
-            )
+            if signal.is_little_endian:
+                least = self.size * 8 - signal.startBit
+                most = least - signal.size
+
+                bits = little[most:least]
+            else:
+                most = signal.startBit
+                least = most + signal.size
 
-        return bitstruct.pack(fmt, *signal_value)
+                bits = big[most:least]
 
-    def decode(self, data, decodeToStr=False):
-        """Return OrderedDictionary with Signal Name: Signal Value
+            unpacked.append(signal.unpack_bitstring(signal.size, signal.is_float, signal.is_signed, bits))
+
+        return unpacked
+
+    def unpack(self, data, report_error=True):
+        """Return OrderedDictionary with Signal Name: object decodedSignal (flat / without support for multiplexed frames)
 
         :param data: Iterable or bytes.
             i.e. (0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xA8)
-        :param bool decodeToStr: If True, try to get value representation as *string* ('Init' etc.)
         :return: OrderedDictionary
         """
+
+        rx_length = len(data)
+        if rx_length != self.size and report_error:
+            print(
+                'Received message 0x{self.id:08X} with length {rx_length}, expected {self.size}'.format(**locals()))
+        else:
+            if sys.version_info > (3,):
+                little, big = self.bytes_to_bitstrings(bytes(data))
+            else:
+                little, big = self.bytes_to_bitstrings(data)
+
+            unpacked = self.bitstring_to_signal_list(self.signals, big, little)
+
+            returnDict= dict()
+
+            for s, v in zip(self.signals, unpacked):
+                returnDict[s.name] = decodedSignal(v, s)
+
+            return returnDict
+
+    def decode(self, data):
```

Comment:
So, this is just `unpack()` with multiplex support?  The names don't really clarify anything to me.

Reply:
yes, it is,
I could imagine that it's good to have a method which decodes just every signal regardless of the multiplexer status.
What your opinion as native speaker: what do you  think is better "decode/encode" or "pack/unpack" ?
Maybe unpack and _unpack_raw?

Reply:
should be probably renamed. 
may be done in another pr.